src/libcsg/modules/io/pdbtopologyreader.cc

```diff
@@ -0,0 +1,415 @@
+    // Read in information from .pdb file
+    ////////////////////////////////////////////////////////////////////////////////
+    int bead_count = 0 ;
+    while ( std::getline(_fl, line) ){
+        if( wildcmp("CRYST1*",line.c_str())){
+             string a, b, c, alpha, beta, gamma;
+             try {
+                                            // 1 -  6       Record name    "CRYST1"
+               a    =string(line,(7-1),9);  // 7 - 15       Real(9.3)      a           (Angstroms)
+               b    =string(line,(16-1),9); //16 - 24       Real(9.3)      b           (Angstroms)
+               c    =string(line,(25-1),9); //25 - 33       Real(9.3)      c           (Angstroms)
+               alpha=string(line,(34-1),7); //34 - 40       Real(7.2)      alpha       (degrees)
+               beta =string(line,(41-1),7); //41 - 47       Real(7.2)      beta        (degrees)
+               gamma=string(line,(48-1),7); //48 - 54       Real(7.2)      gamma       (degrees)
+                                            //56 - 66       LString        Space group
+                                            //67 - 70       Integer        Z value
+            } catch (std::out_of_range& err) {
+              throw std::runtime_error("Misformated pdb file in CRYST1 line");
+            }
+            boost::algorithm::trim(a);
+            boost::algorithm::trim(b);
+            boost::algorithm::trim(c);
+            boost::algorithm::trim(alpha);
+            boost::algorithm::trim(beta);
+            boost::algorithm::trim(gamma);
+	        if ((!wildcmp("90*",alpha.c_str()))||(!wildcmp("90*",alpha.c_str()))||(!wildcmp("90*",alpha.c_str()))){
+	         throw std::runtime_error("Non cubical box in pdb file not implemented, yet!");
+            }
+            double aa = boost::lexical_cast<double>(a)/10.0;
+            double bb = boost::lexical_cast<double>(b)/10.0;
+            double cc = boost::lexical_cast<double>(c)/10.0;
+	        top.setBox(matrix(vec(aa, 0 , 0 ),
+	                          vec(0 , bb, 0 ),
+                              vec(0 , 0 , cc)));
+
+	    }
+        if( wildcmp("CONECT*",line.c_str())){
+            vector<string> bonded_atms;
+            string atm1;
+            // Keep track of the number of bonds
+            int num_bonds = 0;
+            try {
+                // If the CONECT keyword is found then there must be at least
+                // two atom identifiers, more than that is optional. 
+                
+                stringstream ss;
+                ss << string(line.substr(6));
+                // 1 -  6       Record name    "CONECT"
+                //11 -  7       Real(5)        atm1           (ID)
+                ss >> atm1; 
+                string temp_atm;          
+                ss >> temp_atm;
+                bonded_atms.push_back(temp_atm);
+                num_bonds = 1;
+                cout << atm1 << " " << temp_atm << " ";
```

Comment:
is it for debug? why do you output something here?

Reply:
Ooops forgot to remove all the cout statements.